### PR TITLE
[vcpkg-tool-meson] Restore the PKG_CONFIG_PATH environment variable to the correct path

### LINKED ISSUE #40305: vcpkg-tool-meson: PKG_CONFIG_PATH is no longer respected
**Describe the bug**
The PR https://github.com/microsoft/vcpkg/pull/38901 introduced a problem with meson. On my system I have a library installed in `/usr/local` and I'm using a custom `PKG_CONFIG_PATH` to find the library. Prior to 2024.06.15 the library was found by vcpkg.

After some research, the PR added a command line argument to run meson, which should expand the list of additional paths for pkg-config to search. However, it also prevents meson from using the `PKG_CONFIG_PATH`. I verified this by creating a simple meson.build, searching for the library in `/usr/local`, setting the `PKG_CONFIG_PATH` environment variable, and running meson with and without the `--pkg-config-path` argument.

**Environment**
- OS: Linux

**Expected behavior**
Paths in the PKG_CONFIG_PATH environment variable should be respected when calling `vcpkg_configure_meson()`.

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fixes #40305

Paths in the PKG_CONFIG_PATH environment variable should be respected when calling vcpkg_configure_meson(), so remove option `--pkg-config-path` 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~~SHA512s are updated for each updated download.~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

@marcbull @Neumann-A

### CODE REVIEW

**Comment by Neumann-A** [comment]
> Restore the PKG_CONFIG_PATH environment variable to the correct path

You currently do not achieve that. adjust the flag correctly instead of removing it.

**Review by Neumann-A — CHANGES_REQUESTED** [review]

File: ports/vcpkg-tool-meson/portfile.cmake [hunk 1]
```diff
@@ -10,6 +10,11 @@ set(patches
   adjust-args.patch
   remove-freebsd-pcfile-specialization.patch
 )
+
+if(NOT VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_host_path_list(PREPEND ENV{PKG_CONFIG_PATH} "${LIBRARY_PATH}")
```
Comment (Neumann-A) [inline ports/vcpkg-tool-meson/portfile.cmake:15]: why?!?

File: ports/vcpkg-tool-meson/vcpkg_configure_meson.cmake [hunk 2]
```diff
@@ -364,15 +364,15 @@ function(vcpkg_generate_meson_cmd_args)
...
+      vcpkg_list(APPEND arg_OPTIONS "--pkg-config-path;['$ENV{PKG_CONFIG_PATH}','${CURRENT_INSTALLED_DIR}/lib/pkgconfig','${CURRENT_INSTALLED_DIR}/share/pkgconfig']")
       if(VCPKG_TARGET_IS_WINDOWS)
         vcpkg_list(APPEND arg_OPTIONS "-Dcmake_prefix_path=['${CURRENT_INSTALLED_DIR}','${CURRENT_INSTALLED_DIR}/debug','${CURRENT_INSTALLED_DIR}/share']")
       else()
         vcpkg_list(APPEND arg_OPTIONS "-Dcmake_prefix_path=['${CURRENT_INSTALLED_DIR}','${CURRENT_INSTALLED_DIR}/debug']")
       endif()
     elseif(arg_CONFIG STREQUAL "DEBUG")
       vcpkg_list(APPEND arg_OPTIONS -Ddebug=true --prefix "${CURRENT_PACKAGES_DIR}/debug" --includedir ../include)
-      vcpkg_list(APPEND arg_OPTIONS "--pkg-config-path;['${CURRENT_INSTALLED_DIR}/debug/lib/pkgconfig','${CURRENT_INSTALLED_DIR}/share/pkgconfig']")
+      vcpkg_list(APPEND arg_OPTIONS "--pkg-config-path;['$ENV{PKG_CONFIG_PATH}','${CURRENT_INSTALLED_DIR}/debug/lib/pkgconfig','${CURRENT_INSTALLED_DIR}/share/pkgconfig']")
```
[the remark this answers is not in the document]
Comment (JonLiu1993) [reply ports/vcpkg-tool-meson/vcpkg_configure_meson.cmake:375]: I want to get the environment variables of the Linux system and then add its path to the front of `--pkg-config-path`.

File: ports/vcpkg-tool-meson/portfile.cmake (hunk 1 above)
Comment (JonLiu1993) [reply ports/vcpkg-tool-meson/portfile.cmake:15]: I want to get the environment variables of the Linux system and then add its path to the front of `--pkg-config-path`.

Comment (dg0yt) [reply ports/vcpkg-tool-meson/portfile.cmake:15]: `LIBRARY_PATH` is unknown.  
`PREPEND` is wrong.  
System lib dependencies are not so much desired in vcpkg.  
The environment's `PKG_CONFIG_PATH` is *always* wrong for at least one the triplets in cross builds.  
An alternative solution is available.

So does this need to be continued at all?

**Pull request closed without merging**